Assets/MixedRealityToolkit.Extensions/HandPhysicsService/HandPhysicsService.cs

```diff
@@ -0,0 +1,242 @@
+// Copyright(c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Extensions.HandPhysics
+{
+    /// <summary>
+    /// A simple service that creates KinematicRigidbodies on fingertips for physics interactions.
+    /// </summary>
+    [MixedRealityExtensionService(SupportedPlatforms.WindowsUniversal)]
+    public class HandPhysicsService : BaseExtensionService, IHandPhysicsService, IMixedRealityExtensionService
+    {
+        private HandPhysicsServiceProfile handPhysicsServiceProfile;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="name">Friendly name of the service.</param>
+        /// <param name="priority">Service priority. Used to determine order of instantiation.</param>
+        /// <param name="profile">The service's configuration profile.</param>
+        public HandPhysicsService(string name, uint priority, BaseMixedRealityProfile profile) : base(name, priority, profile)
+        {
+            handPhysicsServiceProfile = (HandPhysicsServiceProfile)profile;
+        }
+
+        private IMixedRealityHandJointService HandJointService => handJointService ??
+            (CoreServices.InputSystem as IMixedRealityDataProviderAccess)?.GetDataProvider<IMixedRealityHandJointService>();
+
+        /// <inheritdoc />
+        public GameObject HandPhysicsServiceRoot { get; private set; }
+
+        /// <inheritdoc />
+        public int HandPhysicsLayer { get; set; }
+
+        /// <inheritdoc />
+        public bool UsePalmKinematicBody { get; set; }
+
+        /// <inheritdoc />
+        public GameObject FingerTipKinematicBodyPrefab
+        {
+            get { return fingerTipKinematicBodyPrefab; }
+            set
+            {
+                if (value != null)
+                {
+                    CreateKinematicBodies();
```

Comment:
should we destroy the relevant kinematic bodies in case of value == null? (same goes for palm )